app/controllers/katello/concerns/smart_proxies_controller_extensions.rb

```diff
@@ -0,0 +1,45 @@
+module Katello
+  module Concerns
+    module SmartProxiesControllerExtensions
+      extend ActiveSupport::Concern
+
+      included do
+        append_view_path('app/views/foreman')
+        before_filter :find_resource_and_status, :only => [:pulp_storage, :pulp_status]
+        alias_method_chain :action_permission, :katello
+
+        def pulp_storage
```

Comment:
I am not as familiar with the design of this area (feel free to point me at them), but if I read this correctly, the intent appears to be to add these actions for showing showing status in the smart proxy UI but not to also enable this for retrieval in the API?